src/helper.js

```diff
@@ -537,41 +542,37 @@ class helper {
 
 		let mosaicsFieldObject = [];
 
-		for (const resolvedMosaic of resolvedMosaics) {
-			if (resolvedMosaic.id.toHex() === http.networkCurrency.mosaicId) {
-				mosaicsFieldObject.push(this.networkCurrencyMosaicBuilder(resolvedMosaic));
-			} else {
-				if (0 < mosaicInfos.length && 0 < mosaicNames.length) {
-					let { divisibility } = mosaicInfos.find(info => info.mosaicId === resolvedMosaic.id.toHex());
-
-					mosaicsFieldObject.push({
-						...resolvedMosaic,
-						rawAmount: resolvedMosaic.amount,
-						mosaicId: resolvedMosaic.id.toHex(),
-						amount: helper.formatMosaicAmountWithDivisibility(resolvedMosaic.amount, divisibility),
-						mosaicAliasName: MosaicService.extractMosaicNamespace({ mosaicId: resolvedMosaic.id.toHex() }, mosaicNames)
-					});
-				}
-			}
-		}
+		const uniqueMosaicIds = [...new Set(resolvedMosaics.map(mosaic => mosaic.id.toHex()))];
 
-		return mosaicsFieldObject;
-	}
+		uniqueMosaicIds.forEach(idHex => {
+			const mosaics = resolvedMosaics.filter(mosaic => mosaic.id.toHex() === idHex);
 
-	static networkCurrencyMosaicBuilder = mosaic => {
-		if (!(mosaic instanceof Mosaic))
-			throw new Error('It required Mosaic instance.');
+			const sumAmount = mosaics.reduce((acc, cur) => acc + Number(cur.amount.toString()), 0);
 
-		if (mosaic.id.toHex() !== http.networkCurrency.mosaicId)
-			throw new Error('Mosaic id does not match network Currency.');
+			const mosaicField = {
+				...mosaics[0],
+				rawAmount: UInt64.fromUint(sumAmount),
```

Comment:
if there is precision issue, it's too late to convert to UInt64 here, and you probably want to use BigInt for arithmetic

Reply:
alright, I will take a look at this. 

`amount` from SDK is returned as `UInt64`, `rawAmount` will need to be UInt64 as well.

so we should convert `Uint64` -> `bigInt` and do arithmetic? after that convert back to `Uint64`?

Reply:
to make my point clearer, here is an example:

when you're doing arithmetic on Uint64s in JS, you want to do something like this [or equivalent]
> BigInt(Number.MAX_SAFE_INTEGER) + 2n - BigInt(Number.MAX_SAFE_INTEGER)
> 2

but you're effectively doing arithmetic with Number not BigInt, which could lead to wrong result:
> BigInt(Number.MAX_SAFE_INTEGER + 2) - BigInt(Number.MAX_SAFE_INTEGER)
> 1